kiosk/sources/extensions/kiosk_metadata.move

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// This module defines a `kiosk_metadata` extension 
+// that can be used to add metadata to a kiosk.
+// This metadata can be used to store information
+// about the kiosk, such as a name, an image_url or extra key, value pairs.
+module kiosk::kiosk_metadata {
+
+    use sui::object::{UID, ID};
+    use sui::kiosk::{Self, Kiosk, KioskOwnerCap};
+    use std::string::{String};
+    use sui::vec_map::{Self, VecMap};
+    use sui::dynamic_field as df;
+    use std::vector::{Self};
+    use sui::event::{emit as emit_event};
+
+    struct Metadata has store, copy, drop {
+        metadata: VecMap<String, String>
+    }
+
+    // === Events === 
+    // Event that is emmitted when kiosk metadata is created
+    struct KioskMetaDataCreated has copy, drop {
```

Comment:
Given that events are pruned, I don't think this would make much sense. It's mostly when querying Kiosk you want to show name (or maybe to user when they have multiple Kiosks)